data/scripts.js

```diff
@@ -1028,6 +1204,19 @@ let BattleScripts = {
 	targetTypeChoices(targetType) {
 		return CHOOSABLE_TARGETS.has(targetType);
 	},
+
+	combineResults(left, right) {
+		const NOT_FAILURE = 'string';
+		const NULL = 'object';
+		const resultsPriorities = ['undefined', NOT_FAILURE, NULL, 'boolean', 'number'];
+		if (resultsPriorities.indexOf(typeof left) > resultsPriorities.indexOf(typeof right)) {
+			return left;
+		} else if (left && !right && right !== 0) {
+			return left;
+		} else {
+			return right;
```

Comment:
Why aren't we adding `left + right` when they're both numbers?

Reply:
Because in all the places this function is used right now, it's not actually possible for them both to be numbers, but I can add that functionality back in now if you want.

Reply:
I think it's worth it, just because of the name.

Reply:
I don't want someone to start using it later and get tripped up by it.

Reply:
Oh yeah, now I remember why, because TypeScript doesn't understand that returning a number from this function is OK if both parameters are numbers.

Reply:
I mean, in the worst case there's always `// @ts-ignore`...